src/asm2cfg/asm2cfg.py

```diff
@@ -120,139 +123,255 @@ def get_stripped_and_function_name(line):
+        self.base = base
+        self.offset = offset
 
+    def is_absolute(self):
+        return self.base is None
 
-def get_jump_pattern(stripped, function_name):
-    """
-    Return regexp pattern used to identify jump assembly lines. Support
-    for stripped and non-stripped versions.
+    def is_relative(self):
+        return not self.is_absolute()
 
-    Need to match non-stripped lines:
-    '0x00007ffff7fbf124 <+68>:  jmp  0x7ffff7fbf7c2 <test_function+1762>'
-    and stripped lines:
-    '0x000055555555600f:        jmp  0x55555555603d'
+    def __str__(self):
+        if self.offset is not None:
+            return f'0x{self.abs:x} ({self.base}+{self.offset})'
+        return f'0x{self.abs}'
+
+    def merge(self, other):
+        if self.abs is not None:
+            assert self.abs is None or self.abs == other.abs
+            self.abs = other.abs
+        if self.base is not None:
+            assert self.base is None or self.base == other.base
+            self.base = other.base
+        if self.offset is not None:
+            assert self.offset is None or self.offset == other.offset
+            self.offset = other.offset
+
+
+class Instruction:
     """
-    if stripped:
-        return re.compile(fr'0x0*({HEX_PATTERN}):\W+\w+\W+0x0*({HEX_PATTERN})$')
-    return re.compile(fr'<[+-](\d+)>:.+<{function_name}[+-](\d+)>')
+    Represents a single assembly instruction with it operands, location and
+    optional branch target
+    """
+    def __init__(self, text, lineno, address, opcode, ops, target, imm):  # pylint: disable=too-many-arguments
+        self.text = text
+        self.lineno = lineno
+        self.address = address
+        self.opcode = opcode
+        self.ops = ops
+        self.target = target
+        if imm is not None and (self.is_jump() or self.is_call()):
+            if self.target is None:
+                self.target = imm
+            else:
+                self.target.merge(imm)
 
+    def is_call(self):
+        # Various flavors of call:
+        #   call   *0x26a16(%rip)
+        #   call   0x555555555542
+        #   addr32 call 0x55555558add0
+        return 'call' in self.opcode
 
-def get_unconditional_branch_pattern():
-    """
-    Return regexp pattern used to identify unconditional jumps.
-    """
-    return re.compile(r'jmpq?\b')
+    def is_jump(self):
+        return self.opcode[0] == 'j'
 
+    def is_unconditional_jump(self):
+        return self.opcode.startswith('jmp')
 
-def get_assembly_line_pattern(stripped):
-    """
-    Return regexp pattern used for matching regular assembly lines in the file.
-    Support for stripped and non-stripped versions.
+    def __str__(self):
+        result = f'{self.address}: {self.opcode}'
+        if self.ops:
+            result += f' {self.ops}'
+        return result
+
+    def print_body(self):
+        return self.opcode + ' ' + ','.join(self.ops)
+
+
+def parse_address(line):
+    'Parses leading address of instruction'
```

Comment:
I think it would be more consistent if we use """comment here""" style commenting in this repository. This style is also recommended by Python :)

Reply:
Done.